opm/common/ErrorMacros.hpp

```diff
@@ -40,6 +40,18 @@
 # define OPM_MESSAGE_IF(cond, m) do {} while (false)
 #endif
 
+// macros used to time blocks for example with tracy
+// time block of main part of codes which do not effect performance
+#ifndef OPM_TIMEBLOCK
```

Comment:
Why the `#ifndef` ?

Reply:
> Why the `#ifndef` ?

Enables selectively activating `OPM_TIMEBLOCK`.  Client code may do something along the lines of
```C++
#include <config.h>

#define OPM_TIMEBLOCK(x) \
    SOME_APPROPRIATE_EXPANSION(x)

// ...

#include <opm/common/ErrorMacros.hpp>

// ...
void f()
{
    OPM_TIMEBLOCK("f()");
    // ...
}
```
which will add `SOME_APPROPRIATE_EXPANSION("f()")` to the body of `f()`.  The normal case, without client code providing a definition for `OPM_TIMEBLOCK` will then add nothing to the body of `f()`.  The point is that, with this PR, we can start adding `OPM_TIMEBLOCK` calls where we like without actually incurring any cost.  Developers who then want to do measurements can redefine `OPM_TIMEBLOCK` **before** including `ErrorMacros.hpp` and get targeted measurements.

If they want measurements "everywhere" they might even take steps to have a non-empty definition of `OPM_TIMEBLOCK` in their `config.h` files.